guide/src/reference/attributes/on-js-imports/host_binding.md

```diff
@@ -0,0 +1,139 @@
+# `host_binding`
+
+The `host_binding` attribute is the converse of the [`structural`
+attribute](structural.html). It configures how `wasm-bindgen` will generate JS
+glue to call the imported function. The naming here is intended convey that this
+attribute is intended to implement the semantics of the future [host bindings
+proposal][host-bindings] for WebAssembly.
+
+[host-bindings]: https://github.com/WebAssembly/host-bindings
+[reference-types]: https://github.com/WebAssembly/reference-types
+
+The `host_binding` attribute is intended to be purely related to performance. It
+ideally has no user-visible effect, and well-typed `structural` imports (the
```

Comment:
I think it would be good to specify what "well-typed" means. It actually has little to do with the types, and has a lot more to do with overriding methods.